test/extended/coreos/kdump.go

```diff
@@ -0,0 +1,143 @@
+package coreos
+
+import (
+	"math/rand"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = g.Describe("[sig-coreos][Feature:machine-config-operator][Slow][Serial] kdump", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLIWithoutNamespace("kdump-enablement")
+	g.It("TestKdump", func() {
+		_, workers := clusterNodes(oc)
+		workerNode := workers[rand.Intn(len(workers))]
+
+		config, err := framework.LoadConfig()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dynamicClient := dynamic.NewForConfigOrDie(config)
+
+		mcps := dynamicClient.Resource(schema.GroupVersionResource{
+			Group:    "machineconfiguration.openshift.io",
+			Version:  "v1",
+			Resource: "machineconfigpools",
+		})
+
+		err = kdumpSetup(oc, workerNode, mcps)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = kernelCrash(oc, workerNode, mcps)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = cleanup(oc, workerNode, mcps)
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+})
+
+func kdumpSetup(oc *exutil.CLI, node *corev1.Node, mcps dynamic.NamespaceableResourceInterface) error {
+	// Create 99-infra-kdump-configuration.yaml to reserve the kdump memory
+	// and enable kdump.service via MCO.
+	runCommandAndRetry(createKdumpMC)
+	runCommandAndRetry(createInfraPool)
+
+	// Create infra pool to roll out MC changes
```

Comment:
I'm not familiar with the broader origin tests, so this might be obvious, but...why do we have to create a infra pool here?  Can't we just apply the MC to the worker pool?

Reply:
Oh I think I get it...the infra pool is just to select a single node to run the test on rather than all the worker nodes.